File: Problems/ICPC Asyut Sheets/Sheet #1 (Data type - Conditions)/X. Two intervals.cpp
#include <bits/stdc++.h>
using namespace std;

bool between(int n, int l, int r)
{
    return n >= l && n <= r;
}
int main()
{
    int l1, r1, l2, r2;
    cin >> l1 >> r1 >> l2 >> r2;

    if (l2 > r1 || l1 > r2)
        cout << -1;

    else if (between(l2, l1, r1) && between(r2, l1, r1))
        cout << l2 << " " << r2;
    else if (between(l1, l2, r2) && between(r1, l2, r2))
        cout << l1 << " " << r1;

    else if (between(l2, l1, r1))
        cout << l2 << " " << r1;
    else if (between(l1, l2, r2))
        cout << l1 << " " << r2;

    return 0;
}

File: Problems/ICPC Asyut Sheets/Sheet #1 (Data type - Conditions)/X. Two intervals.py
def between(n, l,r):
    return n >= l and n <= r
l1,r1,l2,r2=map(int,input().split())

if l2 > r1 or l1 > r2: print(-1)
elif between(l2, l1, r1) and between(r2, l1, r1): print(l2,r2)
elif between(l1, l2, r2) and between(r1, l2, r2): print(l1,r1)
elif between(l2, l1, r1): print(l2,r1)
else: print(l1,r2)